scripts/analyze_training_metrics.py

```diff
@@ -0,0 +1,168 @@
+    values_dict: Dict[datetime, float],
+    output_directory: Path,
+) -> bool:
+    sorted_metrics = sorted(values_dict.items())
+    x, y = zip(*sorted_metrics)
+
+    # Compute the threshold
+    if len(y) >= 2:
+        before_last_value = y[-2]
+    else:
+        before_last_value = y[-1]
+    ninety_five_percent = before_last_value * 0.95
+
+    figure = plt.figure()
+    axes = plt.axes()
+
+    # Formatting of the figure
+    figure.autofmt_xdate()
+    axes.fmt_xdata = mdates.DateFormatter("%Y-%m-%d-%H-%M")
+    axes.set_title(f"{model_name} {metric_name}")
+
+    # Display threshold
+    axes.axhline(y=ninety_five_percent, linestyle="--", color="red")
+    plt.annotate(
+        "{:.4f}".format(ninety_five_percent),
+        (x[-1], ninety_five_percent),
+        textcoords="offset points",  # how to position the text
+        xytext=(-10, 10),  # distance from text to points (x,y)
+        ha="center",
+        color="red",
+    )
+
+    # Display point values
+    for single_x, single_y in zip(x, y):
+        label = "{:.4f}".format(single_y)
+
+        plt.annotate(
+            label,  # this is the text
+            (single_x, single_y),  # this is the point to label
+            textcoords="offset points",  # how to position the text
+            xytext=(0, 10),  # distance from text to points (x,y)
+            ha="center",
+        )  # horizontal alignment can be left, right or center
+
+    axes.plot_date(x, y, marker=".", fmt="-")
+    # plt.show(block=True)
+    output_file_path = output_directory.resolve() / f"{model_name}_{metric_name}.svg"
+    LOGGER.info("Saving %s figure", output_file_path)
+    plt.savefig(output_file_path)
+
+    plt.close(figure)
+
+    # Check if the threshold has been crossed
+    return y[-1] < ninety_five_percent
+
+
+def analyze_metrics(metrics_directory: str, output_directory: str):
+    root = Path(metrics_directory)
+
+    metrics = defaultdict(
+        lambda: defaultdict(dict)
+    )  # type: Dict[str, Dict[str, Dict[datetime, float]]]
+
+    threshold_ever_crossed = False
+
+    for metric_file_path in root.glob("metric*.json"):
+
+        # Load the metric
+        with open(metric_file_path, "r") as metric_file:
+            metric = json.load(metric_file)
+
+        # Get the model, date and version from the file
+        # TODO: Might be better storing it in the file
+        file_path_parts = metric_file_path.stem.split("_")
+
+        assert file_path_parts[:5] == ["metric", "project", "relman", "bugbug", "train"]
+        model_name = file_path_parts[5]
+        assert file_path_parts[6:8] == ["per", "date"]
+        date_parts = list(map(int, file_path_parts[8:14]))
+        date = datetime(
+            date_parts[0],
+            date_parts[1],
+            date_parts[2],
+            date_parts[3],
+            date_parts[4],
+            date_parts[5],
+            tzinfo=timezone.utc,
+        )
+        version = file_path_parts[14:]  # noqa TODO: Use version
```

Comment:
Why the noqa?

Reply:
Because it is not used yet and flake8 is emitting a warning

Reply:
Ah, then I'd comment the whole line

Reply:
So we can avoid the noqa